mpf/devices/segment_display/segment_display.py

```diff
@@ -155,11 +169,46 @@ def add_text_entry(self, text, color, flashing, flash_mask, transition, transiti
             raise ValueError(f"Unknown update_method '{self.config['update_method']}' for segment display {self.name}")
 
         # For the replace-text update method, skip the stack and write straight to the display
-        new_text = TextTemplate(self.machine, text).evaluate({})
-        text = SegmentDisplayText.from_str(new_text, self.size, self.config['integrated_dots'],
-                                           self.config['integrated_commas'], self.config['use_dots_for_commas'],
-                                           color)
-        self._update_display(SegmentDisplayState(text, flashing, flash_mask))
+
+        ###############################
+        #Store current color and text as previous text/color of next run even if no transition in this step,
```

Comment:
The linter will complain that you don't have a space between the # and the comment text, please add one and remove the top/bottom block comment lines